### PR TITLE
Add small coin selection improvements

### PR DESCRIPTION
**Pull request by jack60612**
- Add support for a target of 0
- Change type of max_coin_amount to uint64
- Validate if coins are below max_coin_amount for future RPC use.

I would go commit by commit for easier reviewing.

### CODE REVIEW

File: chia/wallet/coin_selection.py [hunk 1]
```diff
@@ -56,6 +56,11 @@ async def select_coins(
             f"Transaction for {amount} is greater than spendable balance of {sum_spendable_coins}. "
             "There may be other transactions pending or our minimum coin amount is too high."
         )
+    if amount == 0 and sum_spendable_coins == 0:
+        raise ValueError(
+            "No coins available to spend, you can not create a coin with an amount of 0,"
+            " without already having coins."
+        )
```
Comment (Quexington) [inline chia/wallet/coin_selection.py:63]: I see the point of this raise, but is there any scenario where we would want to call `select_coins` and have it return an empty list?  It wouldn't hurt to have this in here, but I don't see the point.

File: chia/wallet/coin_selection.py [hunk 2]
```diff
@@ -97,7 +102,7 @@ async def select_coins(
                 coin_set = {greater_coin}
         return coin_set
     else:
-        # if smaller_coin_sum == amount and len(smaller_coins) >= max_num_coins.
+        # if smaller_coin_sum == amount and len(smaller_coins) >= max_num_coins, or amount == 0.
```
Comment (Quexington) [inline chia/wallet/coin_selection.py:105]: This comment seems inconsistent with the line it's trying to reference.  I can't tell if it's incorrect, but it's different.

File: chia/wallet/coin_selection.py (hunk 1 above)
Comment (jack60612) [reply chia/wallet/coin_selection.py:63]: No, as we assert it later, however i want a clean and easy to understand error

File: chia/wallet/coin_selection.py (hunk 2 above)
Comment (jack60612) [reply chia/wallet/coin_selection.py:105]: yeah ill make it clearer

**Review by Quexington — APPROVED** [review]
Seems fine to me, but someone closer to this code might also review it.

File: chia/wallet/coin_selection.py [hunk 3]
```diff
@@ -97,7 +102,7 @@ async def select_coins(
                 coin_set = {greater_coin}
         return coin_set
     else:
-        # if smaller_coin_sum == amount and len(smaller_coins) >= max_num_coins.
+        # if smaller_coin_sum == amount and (len(smaller_coins) >= max_num_coins or amount == 0)
```
Comment (trepca) [inline chia/wallet/coin_selection.py:105]: what is the reason for leaving this comment in?

Comment (jack60612) [reply chia/wallet/coin_selection.py:105]: This improves readability, and makes it easier to understand, Mariano suggested the comment so I think it's a good idea.

**Review by trepca — APPROVED** [review]

**Pull request merged**